server/app.py

```diff
@@ -966,8 +967,24 @@ def payment_intent_succeeded(event):
 
 @celery.task(name="app.customer_subscription_deleted")
 def customer_subscription_deleted(event):
-    subscription = event["data"]["object"]
-    rdo = close_rdo(subscription["id"])
+    subscription = stripe.Subscription.retrieve(event["data"]["object"]["id"], expand=["customer"])
+    customer = subscription["customer"]
+    method = subscription["cancellation_details"].get('comment', 'Staff')
+    reason = subscription["cancellation_details"].get('reason')
+
+    contact = Contact.get(email=customer["email"])
+    rdo = close_rdo(subscription["id"], method=method, contact=contact)
+    text = f"{contact.name}'s ${rdo.amount}/{rdo.installment_period} donation was cancelled due to {reason}"
+    if reason == "cancellation_requested":
+        text += f" ({method})"
```

Comment:
**suggestion (non-blocking):** is reason always `cancellation_requested`?  If so, we could make this a little more succinct/scannable and have it read:

Cancellation: Ivana Cancel $35.00/monthly (Staff) 
-or-
Cancellation: Ivana Cancel $35.00/monthly (Member Portal)

Reply:
No, and that's a good question that leads to some further info. Though not something I can test per se, this update should also end up pushing cancellations due to card failures and the like to slack. I'm assuming that would be good to see in that channel, but let me know if you think it's overkill and I can re-write this to kick those out. I figured it would be better to start with sending any cancellations (whether user/staff initiated or from charging failure) to that channel and if it's too much noise, we can reduce, but I'm interested in your opinion.

Reply:
We talked about this in realtime, but I think this is sound reasoning 🤝